core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java

```diff
@@ -177,7 +187,11 @@ public final T directExecutor() {
 
   @Override
   public final T executor(Executor executor) {
-    this.executor = executor;
+    if (executor != null) {
```

Comment:
That's true, but it also then hides condition that needs testing. I'm in favor of that for checkNotNull, where the case is just a distraction. Or like the previous usages of `firstNonNull()`, since the code was so complex already. I could go either way, but I chose the boring approach that makes it more obvious when we're skipping test cases.